src/main/java/org/skriptlang/skript/log/runtime/RuntimeErrorCatcher.java

```diff
@@ -0,0 +1,118 @@
+package org.skriptlang.skript.log.runtime;
+
+import ch.njol.skript.log.SkriptLogger;
+import com.google.common.collect.ImmutableList;
+import org.skriptlang.skript.log.runtime.Frame.FrameOutput;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.logging.Level;
+
+/**
+ * A {@link RuntimeErrorConsumer} to be used in {@link RuntimeErrorManager} to catch {@link RuntimeError}s.
+ * This should always be used with {@link #start()} and {@link #stop()}.
+ */
+public class RuntimeErrorCatcher implements RuntimeErrorConsumer {
+
+	private List<RuntimeErrorConsumer> storedConsumers = new ArrayList<>();
+
+	private final List<RuntimeError> cachedErrors = new ArrayList<>();
+
+	private final List<Entry<FrameOutput, Level>> cachedFrames = new ArrayList<>();
+
+	public RuntimeErrorCatcher() {}
+
+	/**
+	 * Gets the {@link RuntimeErrorManager}.
+	 */
+	private RuntimeErrorManager getManager() {
+		return RuntimeErrorManager.getInstance();
+	}
+
+	/**
+	 * Starts this {@link RuntimeErrorCatcher}, removing all {@link RuntimeErrorConsumer}s from {@link RuntimeErrorManager}
+	 * and storing them in {@link #storedConsumers}.
+	 * Makes this {@link RuntimeErrorCatcher} the only {@link RuntimeErrorConsumer} in {@link RuntimeErrorManager}
+	 * to catch {@link RuntimeError}s.
+	 * @return This {@link RuntimeErrorCatcher}
+	 */
+	public RuntimeErrorCatcher start() {
+		storedConsumers = getManager().removeAllConsumers();
+		getManager().addConsumer(this);
+		return this;
+	}
+
+	/**
+	 * Stops this {@link RuntimeErrorCatcher}, removing from {@link RuntimeErrorManager} and restoring the previous
+	 * {@link RuntimeErrorConsumer}s from {@link #storedConsumers}.
+	 * Prints all cached {@link RuntimeError}s, {@link #cachedErrors}, and cached {@link FrameOutput}s, {@link #cachedFrames}.
+	 */
+	public void stop() {
+		if (!getManager().removeConsumer(this)) {
+			SkriptLogger.LOGGER.severe("[Skript] A 'RuntimeErrorCatcher' was stopped incorrectly.");
+			return;
+		}
+		getManager().addConsumers(storedConsumers.toArray(RuntimeErrorConsumer[]::new));
+		for (RuntimeError runtimeError : cachedErrors)
+			storedConsumers.forEach(consumer -> consumer.printError(runtimeError));
+		for (Entry<FrameOutput, Level> entry : cachedFrames)
+			storedConsumers.forEach(consumer -> consumer.printFrameOutput(entry.getKey(), entry.getValue()));
+	}
+
+	/**
+	 * Gets all the cached {@link RuntimeError}s.
+	 */
+	public List<RuntimeError> getCachedErrors() {
+		return ImmutableList.copyOf(cachedErrors);
+	}
```

Comment:
```suggestion
	public @Unmodifiable List<RuntimeError> getCachedErrors() {
		return Collections.unmodifiableList(cachedErrors);
	}
```
Better to have an unmodifiable view than a new copy of a list